plugins/modules/foreman_image.py

```diff
@@ -0,0 +1,128 @@
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: foreman_image
+short_description: Manage Foreman Images using Foreman API
+description:
+  - Create and Delete Foreman Images using Foreman API
+author:
+  - "Mark Hlawatschek (@hlawatschek) ATIX AG"
+options:
+  name:
+    description: Image name
+    required: true
+    type: str
+  compute_resource:
+    description: Compute resource the image is assigned to
+    required: true
+    type: str
+  uuid:
+    aliases:
+      - image_uuid
+    description: UUID or Marketplace URN of the operatingsystem image
+    required: true
+    type: str
+  image_username:
+    description: Username that is used to login into the operating system
+    required: true
+    type: str
+  image_password:
+    description: Password that is used to login into the operating system
+    required: false
+    type: str
+  operatingsystem:
+    description: Operating system that will be deployed using the image
+    required: true
+    type: str
+  architecture:
+    description: architecture of the image
+    required: true
+    type: str
+  user_data:
+    description: Image supports user_data
+    required: false
+    type: bool
+extends_documentation_fragment:
+  - foreman
+  - foreman.entity_state
+'''
+
+EXAMPLES = '''
+   - name: create Image for EC2
+     foreman_image:
+        name: CentOS
+        image_uuid: "ami-0ff760d16d9497662"
+        image_username: "centos"
+        operatingsystem: "CentOS 7"
+        compute_resource: "AWS"
+        architecture: "x86_64"
+'''
+
+RETURN = ''' # '''
+
+from ansible.module_utils.foreman_helper import ForemanEntityAnsibleModule
+
+
+class ForemanImageModule(ForemanEntityAnsibleModule):
+    pass
+
+
+def main():
+    module = ForemanImageModule(
+        argument_spec=dict(
+            image_username=dict(required=True),
+            image_password=dict(no_log=True),
+        ),
+        entity_spec=dict(
+            name=dict(required=True),
+            username=dict(type='invisible'),
+            uuid=dict(required=True, aliases=['image_uuid']),
+            password=dict(type='invisible', no_log=True),
+            compute_resource=dict(type='entity', required=True),
+            architecture=dict(type='entity', required=True),
+            operatingsystem=dict(type='entity', required=True),
+            user_data=dict(type='bool')
+        ),
+        entity_scope="compute_resource"
+    )
+
+    entity_dict = module.clean_params()
+
+    with module.api_connection():
+        entity_dict['username'] = entity_dict['image_username']
+        if 'image_password' in entity_dict:
+            entity_dict['password'] = entity_dict['image_password']
+        operatingsystem_id = module.find_operatingsystem(entity_dict['operatingsystem'], thin=True)['id']
+        entity, entity_dict = module.resolve_entities(entity_dict)
```

Comment:
I *think* this will fail if you try to add an image with the same name but different OS to the same CR, as it's uniqueness is defined on the (CR, OS) scope:
https://github.com/theforeman/foreman/blob/26dda8dbbf32d357657c3d22018c9d247fb0cb4e/app/models/image.rb#L12

Can you please add a test for that?

Reply:
Imo the expected behavior would be a modification of the existing image with the same name and cr/name is the identifier.

Reply:
is it? in the UI, I can create an image called "testimage" on CR1 with OS1, and another one with OS2.

Reply:
oh, you're right. This makes it ugly. identifier is cr/name-os then?

Reply:
yeah, I think you need "scope:CR, search:name+OS" to get an unique restult

Reply:
ok, I think I found a solution for this. Could you please re-review ;-) ?

Reply:
Will do. btw, no need to explicitly request another review in this case ;)